lib/Migration/Version1105Date20210922104324.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Mail\Migration;
+
+use Closure;
+use OCP\DB\Exception;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+use Psr\Log\LoggerInterface;
+
+class Version1105Date20210922104324 extends SimpleMigrationStep {
+	protected $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		$qb = $this->connection->getQueryBuilder();
+		$qb->select('accounts.id')
+			->from('mail_accounts', 'accounts')
+			->leftJoin('accounts', 'mail_provisionings', 'provisionings', $qb->expr()->eq('accounts.provisioning_id', 'provisionings.id'))
+			->where($qb->expr()->isNotNull('accounts.provisioning_id'))
+			->andWhere($qb->expr()->isNull('provisionings.id'));
+
+		try {
+			$result = $qb->execute();
+		} catch (Exception $e) {
+			return;
+		}
+
+		$accountIds = array_map(static function ($row) {
+ 			return (int)$row['id'];
+		}, $result->fetchAll());
+		$result->closeCursor();
+
+		$qb = $this->connection->getQueryBuilder();
+		$qb->delete('mail_accounts')
+			->where($qb->expr()->in('id', $qb->createNamedParameter($accountIds, IQueryBuilder::PARAM_INT_ARRAY)));
+
+		try {
+			$qb->execute();
+		} catch (Exception $e) {
+			return;
```

Comment:
can we at least log this? Why do we silently ignore errors?

Reply:
Throwing exceptions in migrations is a bit tricky. It may interrupt the updater process. I'm using a try-catch here because the migration is not super important. If something goes wrong here (which is unlikely) the update should continue. I don't have a strong opinion on this point. We can either log the exception or not use try-catch here (like in our other migrations).

Reply:
Then let's keep the try-catch but log an error :+1: